src/ZODB/BaseStorage.py

```diff
@@ -245,8 +235,7 @@ def tpc_finish(self, transaction, f=None):
         # to send an invalidation message to all of the other
         # connections!
 
-        self._lock_acquire()
-        try:
+        with self._lock:
             if transaction is not self._transaction:
                 raise POSException.StorageTransactionError(
                     "tpc_finish called with wrong transaction")
```

Comment:
It gets released in the finally below.

(I may be misunderstanding your question.)